core/field_textinput.js

```diff
@@ -175,6 +178,36 @@ Blockly.FieldTextInput.prototype.showEditor_ = function(opt_quietInput, opt_read
   Blockly.FieldTextInput.htmlInput_ = htmlInput;
   div.appendChild(htmlInput);
 
+  if (opt_withArrow) {
+    // Move text in input to account for displayed drop-down arrow.
+    if (this.sourceBlock_.RTL) {
+      htmlInput.style.paddingLeft = (this.arrowSize_+ Blockly.BlockSvg.DROPDOWN_ARROW_PADDING) + 'px';
+    } else {
+      htmlInput.style.paddingRight = (this.arrowSize_ + Blockly.BlockSvg.DROPDOWN_ARROW_PADDING) + 'px';
+    }
+    // Create the arrow.
+    var dropDownArrow =
+        goog.dom.createDom(goog.dom.TagName.IMG, 'blocklyTextDropDownArrow');
+    dropDownArrow.setAttribute('src',
+      Blockly.mainWorkspace.options.pathToMedia + 'dropdown-arrow-dark.svg');
+    dropDownArrow.style.width = this.arrowSize_ + 'px';
+    dropDownArrow.style.height = this.arrowSize_ + 'px';
+    dropDownArrow.style.top = this.arrowY_ + 'px';
+    dropDownArrow.style.cursor = 'pointer';
+    // Magic number for positioning the drop-down arrow on top of the text editor.
+    var dropdownArrowMagic = '11px';
+    if (this.sourceBlock_.RTL) {
+      dropDownArrow.style.left = dropdownArrowMagic;
+    } else {
+      dropDownArrow.style.right = dropdownArrowMagic;
+    }
+    if (opt_arrowCallback) {
+      htmlInput.dropDownArrowMouseWrapper_ = Blockly.bindEvent_(dropDownArrow,
```

Comment:
arrowCallback needs to call Blockly.Touch.clearTouchIdentifier, unless you want to bind to mouseup here to make sure you end the gesture.